pkg/cri/containerd.go

```diff
@@ -232,23 +232,47 @@ func (c *Containerd) createContainer(image string,
 
 		// pull the v3io-fuse image
 		// [IG-23016] MountVolume.SetUp failed for volume storage in k8s 1.29
+		//
 
-		var ctrPath string
+		var err error
 
-		// Check if "/usr/local/bin/ctr" exists
-		if _, err := os.Stat("/usr/local/bin/ctr"); err == nil {
+		// Get path to ctr
+		var ctrPath string
+		if ctrPath, err = exec.LookPath("ctr"); err == nil {
+		} else if _, err = os.Stat("/usr/local/bin/ctr"); err == nil {
 			ctrPath = "/usr/local/bin/ctr"
-		} else if _, err := os.Stat("/usr/bin/ctr"); err == nil {
+		} else if _, err = os.Stat("/usr/bin/ctr"); err == nil {
 			ctrPath = "/usr/bin/ctr"
-		} else {
+		}
+		if err != nil {
 			// Return an error if neither file exists
 			journal.Error("Failed to pull image: ctr not found",
 				"containerName", containerName,
 				"image", image)
 			return nil, err
 		}
 
-		cmd := exec.Command(ctrPath, "-n", "k8s.io", "images", "pull", "--hosts-dir", "/etc/containerd/certs.d/", image)
```

Comment:
continue here if image starts with : `gcr.io/iguazio` - continue as usual, no need to assume aws or anything

if image has `aws_account_id.dkr.ecr.**region**.amazonaws.com` then it is ecr and region is specified

if image starts with acr.com/moshe , continue as usual